controllers/dashboard_controller.go

```diff
@@ -559,31 +566,43 @@ func (r *GrafanaDashboardReconciler) getDashboardModel(cr *v1beta1.GrafanaDashbo
 	return dashboardModel, fmt.Sprintf("%x", hash.Sum(nil)), nil
 }
 
-func (r *GrafanaDashboardReconciler) Exists(client *grapi.Client, uid string, title string, folderID int64) (bool, string, error) {
-	dashboards, err := client.Dashboards()
-	if err != nil {
-		return false, "", err
-	}
+func (r *GrafanaDashboardReconciler) Exists(client *genapi.GrafanaHTTPAPI, uid string, title string, folderUID string) (bool, string, error) {
+	tvar := "dash-db"
+
+	page := int64(1)
+	limit := int64(1000)
```

Comment:
will this only return up to a maximum of 1000 dashboards?

Reply:
This limit is just the single page size
Below that, we have a loop fetching pages until we run out of results